packages/query/src/bitcoin/ordinals/inscriptions.hooks.ts

```diff
@@ -40,3 +41,52 @@ export function useNumberOfInscriptionsOnUtxo({
     [inscriptions]
   );
 }
+
+export function useInscriptions({
+  taprootKeychain,
+  nativeSegwitAddress,
+}: {
+  nativeSegwitAddress: string;
+  taprootKeychain: HDKey | undefined;
```

Comment:
Any other ideas how we could do this? 

Here, we're passing the keychain to `@leather-wallet/queries` to derive addresses. Which, seems like a responsibility that shouldn't belong to this package.

It's tricky, because we can't just pass an array of string, as we need to recurse over. 

What if we pass a callback that generates a new address? `getNextTaprootAddress(index: number)`? For consistency we could do the same with nativeSegwit addresses too.